### ISSUE TITLE
Can not connect to postgres from remote host -- timescaledb-single

### ISSUE THREAD

**Issue by allllan** [timescale/timescaledb-kubernetes#232]
This is likely not a bug, but more my lack of understanding of the setup... I've deployed timescaledb-single to our k8s cluster and I'm able to connect to the database from within a pod using psql. I can telnet to the service's external IP on port 5432 from our master node, but I can not connect to the db using psql from a remote machine. I'm getting Network is unreachable error.

I deployed the timescaledb-multinode version and am able to connect to the db from remote hosts just fine. I'm sure it's something obvious, but I'm not able to figure out what I'm doing wrong.

Also, is SSL a must for the timescaledb-single version? I tried setting postgresql.parameters.ssl to 'off' in my values file, but that seems to break the deployment. How would one go about disabling SSL for the single version?

kind regards...

**Comment by throrin19** [comment]
Hello,

I don't know if I respond to your question but, the server is only set to accept ssl connections in the default pg_hba. To change this, You should edit the configuration directly in your value.yaml like this : 

```yml
patroni:
  postgresql:
    pg_hba:
    - local     all             all                                   peer
    - hostssl   all             all                127.0.0.1/32       md5
    - hostssl   all             all                ::1/128            md5
    - hostssl   replication     standby            all                md5
    - hostssl   all             all                all                md5
    - host      all             all                all                md5
```

> **warning** : If you use pgBouncer, at this time, it's not possible to edit it's pg_hba

**Comment by feikesteenbergen** [comment]
> 

Also, is SSL a must for the timescaledb-single version? I tried setting postgresql.parameters.ssl to 'off' in my values file, but that seems to break the deployment. How would one go about disabling SSL for the single version?

Yeah, like @throrin19 says, the `pg_hba.conf` still has `hostssl` configuration lines. As long as those are in there, you cannot start PostgreSQL.

First, why do you want to remove SSL? What purprose does it serve?

Second, if you must, you can remove SSL support, by removing the lines from the `pg_hba.conf` that mention `hostssl`. For example, you could do:

```yaml
patroni:
  postgresql:
    pg_hba:
    - local  all             all                                   peer
    - host   all             all                127.0.0.1/32       md5
    - host   all             all                ::1/128            md5
    - host   replication     standby            all                md5
    - host   all             all                all                md5
```

In `pg_hba.conf` a `host` line will match either an ssl or a non-ssl connection attempt.

**Comment by thinairwireless** [comment]
Turning off hostssl in pg_hba proved challenging for us as well, but not as challenging as getting SSL working on an already secure Azure VNET service endpoint (internal network). Yes, we got this running on AKS... there is also a bit of a performance benefit of not having to do SSL decrypt/decompress on an already secure network.  So, here's how we disabled SSL on a  deployed timescaledb single cluster (not sure if this is the best way, but it works):

1) Update pgbouncer ConfigMap pg_hba to desired setting (we found much easier with pgbouncer for 10.0.0.0/8):
[hba.txt](https://github.com/timescale/timescaledb-kubernetes/files/5456039/hba.txt)
2) Login to pgbouncer using Admin console within cluster (http://www.pgbouncer.org/usage.html#admin-console):
psql -p 6432 -U pgbouncer -d pgbouncer
3) Issue a PAUSE; RELOAD; or SHUTDOWN; command within the console
4) Restart cluster using patronictl

This effectively changed pg_hba for pgbouncer and allowed us to connect to timesecaledb on an internal (secure) network and still provide SSL for external load balancer/ingress.  We did this on master and may need this on replicas as we did see this hba change revert back when testing HA.
